src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Administration/GeneralDataProvider.php

```diff
@@ -0,0 +1,95 @@
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Configure\AdvancedParameters\Administration;
+
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Form\FormDataProviderInterface;
+use PrestaShopBundle\Form\Exception\DataProviderError;
+use PrestaShopBundle\Form\Exception\DataProviderErrorCollection;
+use PrestaShopBundle\Form\Exception\DataProviderException;
+
+/**
+ * This class is responsible of managing the data manipulated using general form
+ * in "Configure > Advanced Parameters > Administration" page.
+ */
+final class GeneralDataProvider implements FormDataProviderInterface
+{
+    /**
+     * @var DataConfigurationInterface
+     */
+    private $dataConfiguration;
+
+    public function __construct(
+        DataConfigurationInterface $dataConfiguration
+    ) {
+        $this->dataConfiguration = $dataConfiguration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData()
+    {
+        return $this->dataConfiguration->getConfiguration();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setData(array $data)
+    {
+        $this->validate($data);
+
+        return $this->dataConfiguration->updateConfiguration($data);
+    }
+
+    /**
+     * Perform validations on form data.
+     *
+     * @param array $data
+     */
+    private function validate(array $data): void
+    {
+        $errors = new DataProviderErrorCollection();
+        $frontOfficeLifeTimeCookie = $data[GeneralType::FIELD_FRONT_COOKIE_LIFETIME];
+        $backOfficeLifeTimeCookie = $data[GeneralType::FIELD_BACK_COOKIE_LIFETIME];
+
+        if (!is_numeric($frontOfficeLifeTimeCookie) || $frontOfficeLifeTimeCookie < 0) {
+            $errors->add(new DataProviderError(FormDataProvider::ERROR_NOT_NUMERIC_OR_LOWER_THEN_0, GeneralType::FIELD_FRONT_COOKIE_LIFETIME));
+        }
+
+        if (!is_numeric($backOfficeLifeTimeCookie) || $backOfficeLifeTimeCookie < 0) {
+            $errors->add(new DataProviderError(FormDataProvider::ERROR_NOT_NUMERIC_OR_LOWER_THEN_0, GeneralType::FIELD_BACK_COOKIE_LIFETIME));
+        }
+
+        if (!$errors->isEmpty()) {
```

Comment:
Before data provider would always return errors, here I am throwing exception with those errors inside the exception. 

On one hand, I think it makes way more sense to throw exception and catch is when something goes wrong, rather then returning array of errors.

On the other this requires exception with additional param, I am not sure if that's a bad thing. Also this means that that I am changing the behaviour that exists in other options form. As part of form simplification I could change other option forms to behave this way as well, but this is a big change so I need other oppinions before I proceed with it. 

Also if we stick with returning array of errors, should I return array of DTO's or a Collection? While I like collection more, using collection breaks general behaviour that might at this point be expected of options forms.

Reply:
I feel like
```
throw new DataProviderException('Administration general data is invalid', 0, null, $errors);
```
is actually quite a nice compromise 👍  between the "throw exception" rule and the "gather all errors" behavior

I like it like this, but I'd be happy to hear more opinions. @jolelievre @zuk3975 ?

Reply:
If I got it right, the problem actually comes from allowing to set these invalid values in configuration in first place, but we are trying to validate something in a form DataProvider (a "read" operation) after its already wrong in db?
So if you ask me it seems all conceptually wrong (but I suppose we don't have any better way to handle this?)

Anyway, for mee too - throwing exception with a list of errors seems better than passing them around manually :+1: